pennylane/variable.py

```diff
@@ -83,23 +73,24 @@ class Variable:
 
     Args:
         idx  (int): index into the value vector, >= 0
-        name (None, str): name of the argument
+        name (str): structured name of the parameter
+        dict_key (None, str): for auxiliary parameters the name of the base parameter, otherwise None
```

Comment:
```suggestion
        name (str): structured name of the parameter (e.g. 'weights[0][2]')
        dict_key (None, str): for auxiliary parameters the name of the base parameter (e.g. 'weights'), otherwise None
```